api/jobset/v1alpha2/jobset_types.go

```diff
@@ -46,6 +46,10 @@ type JobSetSpec struct {
 	// +listMapKey=name
 	ReplicatedJobs []ReplicatedJob `json:"replicatedJobs,omitempty"`
 
+	// Network defines the networking options for the jobset.
+	// +optional
```

Comment:
I missed this, we need to make sure it is immutable, please add this (see SuccessPolicy below as an example):

// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"

Reply:
oh cool - TIL there is an annotation to ensure that something is mutable! I definitely have a few places this would be handy.

Reply:
this is specific to kubebuilder